### ISSUE TITLE
Private Registry Error: x509: certificate signed by unknown authority

### ISSUE THREAD

**Issue by pcgeek86** [docker/docker#25587]
# Scenario

## Registry

- We set up a private Docker Registry, using VMware Harbor, secured with a DigiCert TLS certificate
- We are able to access the registry from a web browser using properly validated TLS

## Docker Host

- We set up a separate Docker 1.12 Host on Ubuntu 14.04
- When we use `docker login myregistry.domain.com` to authenticate to the registry, we receive: `x509: certificate signed by unknown authority`
- The DigitCert root & intermediate certificate authority's public keys are in the `/etc/ssl/certs` directory

## Miscellaneous

- Using the `--insecure-registry` option is not the desired solution
- We have already attempted many variations of using the `REGISTRY_HTTP_TLS_CERTIFICATE` environment variable to instruct the Docker Engine to trust our root and intermediate certificate authorities

What is the proper method of instructing the Docker Engine to trust our certificate that was issued from a trusted root CA? The documentation mentions:

> Note: In the absence of any root certificate authorities, Docker will use the system default (i.e., host’s root CA set).

![image](https://cloud.githubusercontent.com/assets/466713/17570275/f7fc1bdc-5f08-11e6-82e6-33e89aedbb6d.png)

However, this does not seem to be the case. It seems like there's a bug here.

Cheers,
**Trevor Sullivan**
Docker Captain
Microsoft MVP: Cloud & Data Center Management
https://trevorsullivan.net
https://twitter.com/pcgeek86

**Comment by dmcgowan** [comment]
Do you have any certs in `/etc/docker/certs.d/myregistry.domain.com`? Currently Docker will not use by custom certs defined in `/etc/docker/certs.d` and system certificates, this will be fixed in a later version of docker using an upstream change in golang.

**Comment by pcgeek86** [comment]
We tried placing the cert chain into `/etc/docker/certs.d/my.domain.com`, but that didn't seem to work. What we eventually discovered, after significant trial and error, is that our NGINX instance in VMware Harbor was missing the intermediate CA certificate. After we added this, things started working. 

Looking forward to the improvements to TLS in future Docker versions! Thanks @dmcgowan @thaJeztah
